modules/gitops-argo-rollouts-architecture-overview.adoc

```diff
@@ -0,0 +1,67 @@
+// Module included in the following assemblies:
+//
+// * argo_rollouts/using-argo-rollouts-for-progressive-deployment-delivery.adoc
+
+:_mod-docs-content-type: CONCEPT
+[id="gitops-argo-rollouts-architecture-overview_{context}"]
+= Argo Rollouts architecture overview
+
+Argo Rollouts support is enabled on a cluster by installing the {gitops-title} Operator and configuring a `RolloutManager` custom resource (CR) instance.
+
+After a RolloutManager CR is created, the {gitops-title} Operator installs Argo Rollouts into that same namespace. This step includes the installation of the Argo Rollouts controller and the resources required for handling Argo Rollouts, such as CRs, roles, role bindings, and configuration data.
+
+The Argo Rollouts controller can be installed in two different modes:
+
+* *Cluster-scoped mode* (default): The controller oversees resources throughout all namespaces within the cluster.
+* *Namespace-scoped mode*: The controller monitors resources within the namespace where Argo Rollouts is deployed.
+
+The architecture of Argo Rollouts is structured into components and resources. Components are used to manage resources. For example, the AnalysisRun controller manages the `AnalysisRun` CR.
+
+Argo Rollouts include several mechanisms to gather analysis metrics to verify that a new application version is deployed:
+
+* *Prometheus metrics*: The `AnalysisTemplate` CR is configured to connect to Prometheus instances to evaluate the success or failure of one or more metrics.
+* *Kubernetes job metrics*: Argo Rollouts support the Kubernetes `Job` resource to run analysis on resource metrics. You can verify a successful deployment of an application based on the successful run of Kubernetes jobs.
+
+[id="gitops-argo-rollouts-components_{context}"]
+== Argo Rollouts components
+
+Argo Rollouts consists of several components that enable users to practice progressive delivery in {OCP}.
+
+.Argo Rollouts components
+[options="header"]
+|===
+|Name |Description
+|Argo Rollouts controller |The Argo Rollouts Controller is an alternative to the standard `Deployment` resource and coexists alongside it. This controller only responds to changes in the Argo Rollouts resources and manages the `Rollout` CR. The Argo Rollouts Controller does not modify standard deployment resources
+|AnalysisRun controller |The AnalysisRun controller manages and performs analysis for `AnalysisRun` and `AnalysisTemplate` CRs. It connects a rollout to the metrics provider and defines thresholds for metrics that determine if a deployment update is successful for your application.
+|`Experiment controller` | The `Experiment` controller runs analysis on short-lived replica sets, and manages the `Experiment` custom resource. The controller can also be integrated with `Rollout` resource by specifying the `experiment` step in the canary deployment `strategy` field.
```

Comment:
```suggestion
|`Experiment controller` | The `Experiment` controller runs analysis on short-lived replica sets, and manages the `Experiment` custom resource. The controller can also be integrated with `Rollout` resource by specifying the `Experiment` step in the canary deployment `strategy` field.
```

Reply:
Not sure if needs to be capitalized but wanted to point it out just in case.

Reply:
Fixed. Yeah, this should be capitalized. Nice catch.